### ISSUE TITLE
double upvote on the blockchain.

### ISSUE THREAD

**Issue by zinovi** [steemit/steem#1596]
I was playing with the dtube today, than this happened : 

https://steemd.com/tx/4741f9176889c5bd0d9478722eb5de678835b4c2
https://steemd.com/tx/61ca99077a2a86e9781345e768688a1609b8ceac

As you can see at https://steemd.com/@zinovi that is a double upvote.

Is this the expected behaviour ?

**Comment by mvandeberg** [comment]
It will not count as a second upvote. Subsequent upvotes on the same content count as vote changes.

The current behavior for a vote change is that the number of rshares must change. rshares are voting power times vote weight time steem power. Because your voting power decreased from the first vote to the second, the rshares changes and it counts as a second vote.

We could change this behavior to require that the vote weight changes, but I think this is a fairly minor quirk. I would contact the dtube devs to inform them that their site is creating extra votes and burning through your voting power twice as fast.

**Comment by zinovi** [comment]
I am in constant communication with the dtube team on their private discord.

I've reported the issue to them when it happened and try to keep them updated on the progress of my investigation.

As of now, they are aware of the problem and https://steemit.com/introduceyourself/@hightouch is responsible for fixing it.

I was able to reproduce it and have more information on why it is happening.

steemd doesn't allow for multiple votes on the same block, to the same witness, but doesn't check if a vote with the same weight, for the same post, was already made on a prior block.

That means, that one could make multiple identical votes if they are spaced at least 3 seconds apart.
